sdk/spring/spring-cloud-azure-service/src/test/java/com/azure/spring/cloud/service/implementation/identity/api/AzureAuthenticationTemplateTest.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.identity.api;
+
+import com.azure.spring.cloud.service.implementation.identity.AzureAuthenticationTemplate;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.util.ReflectionTestUtils;
+import java.time.Duration;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class AzureAuthenticationTemplateTest {
+
+    private static final String IS_INITIALIZED = "isInitialized";
+
+    @Test
+    void initCalledOnlyOnce() {
+        AzureAuthenticationTemplate template = new AzureAuthenticationTemplate();
+        AtomicBoolean isInitialized = (AtomicBoolean) ReflectionTestUtils.getField(template, IS_INITIALIZED);
+        assertFalse(isInitialized.get());
+        Properties properties = new Properties();
+        template.init(properties);
+        assertTrue(isInitialized.get());
+        assertFalse(isInitialized.compareAndSet(false, true));
+    }
+
+    @Test
+    void shouldCallInit() {
```

Comment:
I'd say just check the exception instead of this reflecting thing, more from a API user's perspective